### PR TITLE
Day7 assignment- Implemented Routing using react router and fetched data using github api

### PR DESCRIPTION
**Pull request by rasika260896**

### CODE REVIEW

File: Day7-Assignment/src/Components/AboutUs.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const AboutUs =()=>{
```
Comment (ap221882) [inline Day7-Assignment/src/Components/AboutUs.js:1]: Can name your "Components" folder as "components". Keeps the Component folders distinguishable from normal folders.

File: Day7-Assignment/src/Components/BodyComponent.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import CardContainer from './CardContainer';
+import {useState,useEffect} from 'react'
+import SearchBar from './SearchBar';
```
Comment (ap221882) [inline Day7-Assignment/src/Components/BodyComponent.js:4]: Import these(useEffect, useState) with react only in the first line

File: Day7-Assignment/src/Components/BodyComponent.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+           "akshaymarch7",
+           "karanmehta",
+           "ab-rai",
+           "ankita1696"
+       ]
+       useEffect(()=>{
+           userData();
+       },[]);
+   
+    /*function to fetch github api */
```
Comment (ap221882) [inline Day7-Assignment/src/Components/BodyComponent.js:22]: Use multiline comments as 
```
/**
 * 
 */
```
are more readable then.

File: Day7-Assignment/src/Components/CardComponent.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Link } from "react-router-dom"
+
+ const CardComponent = ({item}) =>{
```
Comment (ap221882) [inline Day7-Assignment/src/Components/CardComponent.js:4]: can destructure it like:
``{item: {name, avatar_url and so on}}``

File: Day7-Assignment/src/Components/ErrorComponent.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import {useRouteError} from 'react-router-dom'
+
+const ErrorComponent = () =>{
+    //useRouteError is the hook provided by react-router-dom to get the status or msg from the backend if something went wrong
+    const errorData = useRouteError();
+    return(
+        <>
+        <h1>Something wrong happened</h1>
+        <p>{errorData.statusText || errorData.msg }</p>
```
Comment (ap221882) [inline Day7-Assignment/src/Components/ErrorComponent.js:9]: It is errorData.message

File: Day7-Assignment/src/Components/SearchBar.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import {useState} from 'react'
+
+/**
+	 * @description: function is called on Submit of the form, This function describes the functionality of Search
+	 * @param1: searchText : entered search text
+     * @param2: user:props passed to the function to provide user data
```
Comment (ap221882) [inline Day7-Assignment/src/Components/SearchBar.js:7]: Liked it. also add @return statement and datatype

File: Day7-Assignment/src/index.html [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" type="text/css" href="style.css">
+    <title>Hello World with HTML</title>
+</head>
+<body>
+<div id="root">Not rendered!!!</div>
+<script src="App.js" type="module"></script>
```
Comment (ap221882) [inline Day7-Assignment/src/index.html:12]: Can put this in the head tag. Modules are deferred by default.

File: Day7-Assignment/src/index.html [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" type="text/css" href="style.css">
```
Comment (ap221882) [inline Day7-Assignment/src/index.html:7]: import your css in App module

File: Day7-Assignment/src/style.css [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+#search-bar{
+    width: 300px;
+    margin: 10px;
+    padding: 10px;
+    border-radius: 5px;
+}
+
+form{
```
Comment (ap221882) [inline Day7-Assignment/src/style.css:49]: Keep *, body, html tag selectors on top of the CSS code to follow CSS specificity.

**Review by ap221882 — COMMENTED** [review]
Nicely written. Kudos Rasika!!

File: Day7-Assignment/src/Components/AboutUs.js [hunk 10]
```diff
@@ -0,0 +1,9 @@
...
+    return(
+        <div class="about-us">
+        <h1>Welcome to our Team</h1>
+        <p>We are team of 10 members working in different domains, and We are here to learn React from scratch.</p>
+        </div>
+    )
```
Comment (gk9848970) [inline Day7-Assignment/src/Components/AboutUs.js:7]: Try to proper indent html whenever possible.

File: Day7-Assignment/src/Components/SearchBar.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+const searchMemeber = (searchText,user) =>{
+ return user.filter((item)=>{ return item?.name.toLowerCase().includes(searchText.toLowerCase())})
+}
+
+/**
+	 * @description: SearchBar component consist of input element and its functioality
+	 * @param1: user : user data
+     * @param2: SetFilteredUser : this function sets the value of the filteredUser
+*/
```
Comment (gavandivya) [inline Day7-Assignment/src/Components/SearchBar.js:18]: Appreciated

File: Day7-Assignment/src/Components/AboutUs.js (hunk 10 above)
Comment (rasika260896) [reply Day7-Assignment/src/Components/AboutUs.js:7]: fixed

File: Day7-Assignment/src/App.js [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+            {
+                path: "/search",
+                element: <BodyComponent />,
+                errorElement: <ErrorComponent />
+            },
+            {
+                path: "/about-us",
+                element: <AboutUs />,
+                errorElement: <ErrorComponent />
+            },
```
[the remark this answers is not in the document]
Comment (rasika260896) [reply Day7-Assignment/src/App.js]: got it.fixed

File: Day7-Assignment/src/Components/AboutUs.js (hunk 1 above)
Comment (rasika260896) [reply Day7-Assignment/src/Components/AboutUs.js]: > Nicely written. Kudos Rasika!!

File: Day7-Assignment/src/Components/ErrorComponent.js (hunk 5 above)
Comment (rasika260896) [reply Day7-Assignment/src/Components/ErrorComponent.js]: fixed

File: Day7-Assignment/src/style.css (hunk 9 above)
Comment (rasika260896) [reply Day7-Assignment/src/style.css]: fixed

**Pull request merged**